assets/js/components/app.js

```diff
@@ -42,6 +42,54 @@
       $(document).trigger('app:bind:check_form_validity');
     },
 
+    requirementsExists: function (lessonProgress, cb, cbNotExists) {
+      if(!lessonProgress) {
+        return false;
+      }
+
+      var requirementsElements = $('.lesson-list-panel [data-requirements]').filter(function (index, item) {
+        return $(item).data('requirements').length > 0;
+      });
+
+      var requirementsUnified = requirementsElements.map(function(idx, item) {
+        return {
+          item: $(item),
+          requirements: $(item).data('requirements')
+        }
+      }).toArray();
+
+      var exists = _.find(requirementsUnified, { requirements: [{ content_id: lessonProgress.lesson_id}] });
+
+      if (exists) {
+        var $item = exists.item;
+        if (cb) {
+          return cb($item, lessonProgress.lesson_id);
+        }
+      } else {
+        if (cbNotExists) {
+          return cbNotExists();
```

Comment:
@mandado então manow,melhor vc jogar isso em um dos arquivos q usam, pq tipo, app.js é apenas o unity of work sacou?